packages/edge-login-ui-rn/src/actions/CreateAccountActions.js

```diff
@@ -34,31 +34,36 @@ export function checkUsernameForAvailabilty(data: string) {
     const context = imports.context
     // dispatch(openLoading()) Legacy dealt with state for showing a spinner
     // the timeout is a hack until we put in interaction manager.
-    setTimeout(() => {
-      context
-        .usernameAvailable(data)
-        .then(async response => {
-          if (response) {
+    return new Promise<void>(resolve =>
+      setTimeout(() => {
```

Comment:
Are you sure the "setTimeout" is still needed with your changes?

Reply:
I just improve this action to return a promise, I think I don't resolve the comment above, but I can try to remove it and see what happens next

Reply:
ok, works fine, removed it